File: codes/3.ProgrammingLanguage/编程语言试水/Score100WordsAnalyzer.rs
use std::{fs::read_to_string, time::Instant};

fn main() {
    let start = Instant::now();
    let dictionary: String = read_to_string("Dictionary.txt").unwrap();
    let score_100_words: Vec<&str> = dictionary
        .lines()
        .filter(|line|
            !line.contains("-")
                && !line.contains(" ")
                && !line.contains("="))
        .filter(|word|
            word.chars()
                .map(|c| c as i32 - 'a' as i32 + 1)
                .sum::<i32>() == 100)
        .collect();
    println!("{:?}", score_100_words);
    println!("\nwords number: {}", score_100_words.len());
    println!("        time: {}ms", start.elapsed().as_millis());
}

File: codes/3.ProgrammingLanguage/编程语言试水/Score100WordsAnalyzer.rb
start = Time.now
score100_words = File.readlines("Dictionary.txt")
    .map(&:chomp)
    .select do |line| !line.include?("-") && !line.include?(" ") && !line.include?("=") end
    .select do |line|
        score_sum = 0
        line.chars.each do |c| score_sum += c.ord - 'a'.ord + 1 end
        score_sum == 100
    end
puts score100_words.join(",")
puts "\nwords number: #{score100_words.size}"
puts "        time: #{((Time.now - start) * 1000).round(0)}ms"
